core/src/main/scala/io/finch/RequestReaders.scala

```diff
@@ -135,6 +136,18 @@ trait RequestReaders {
   val binaryBody: RequestReader[Array[Byte]] = binaryBodyOption.failIfNone
 
   /**
+   * A [[RequestReader]] that reads a required chunked streaming binary body, interpreted as a `AsyncStream[Buf]`.
+   */
+  def asyncBody(chunkSize: Int): RequestReader[AsyncStream[Buf]] = {
```

Comment:
The thing is that there is already a `Buf` (well, it's actually `ChannelBuffer`) read from the chunk by Netty. So it's already in memory and we actually can reference it from here w/o copying.